subsys/mgmt/settings/src/settings_mgmt.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <logging/log.h>
+#include <mgmt/mgmt.h>
+#include <mgmt/settings_mgmt.h>
+#include <cborattr/cborattr.h>
+#include <stats/stats.h>
+
+LOG_MODULE_REGISTER(mgmt_settings, CONFIG_MGMT_SETTINGS_LOG_LEVEL);
+
+#define SETTINGS_MGMT_ID_ERASE 6
+
+static int settings_erase(struct mgmt_ctxt *ctxt)
+{
+	int rc = -EFAULT;
+	CborError cbor_err = 0;
+
+#ifdef PM_SETTINGS_STORAGE_ID
+	const struct flash_area *fa;
+	while (1) {
```

Comment:
Is there a reason not to used delayed work or some other scheduled mechanism which is not as blocking? To me it seems like at issue using `while (1)` since the command typically arrives from BLE, we are doing flash operations inside the loop, and BLE has concurrency issues with flash operations.

Reply:
`while` is not used for blocking just easier error handling instead of the
```
if error
  goto out_label
out_label:
  handle error
```

Any concurrency issues should be handled in `flash_area` IMO.

Reply:
Removed when put into a shared function.